sandy/core/cov.py

```diff
@@ -370,8 +370,7 @@ def _reduce_size(self):
         --------
         >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
         >>> non_zero_index, reduce_matrix = S._reduce_size()
-        >>> non_zero_index
-        array([0, 1, 2], dtype=int64)
+        >>> assert non_zero_index.all() == np.array([0, 1, 2]).all()

```

Comment:
here you are not doing what you think you are doing...this is not checking if values are equal.
if you print `non_zero_index.all()` you just geta  True or a False, independently on `np.array([0, 1, 2]).all()`.
The you are just chekcing if `True==True` or `False==False`